test/commons/matches/from-definition.js

```diff
@@ -0,0 +1,209 @@
+          foo: matcher
+        }
+      }))
+    })
+    assert.isFalse(fromDefinition(fixture.firstChild, {
+      attributes: {
+        foo: 'baz'
+      }
+    }))
+  })
+
+  it('matches a definition with an `properties` property', function () {
+    fixture.innerHTML = '<input />'
+    const matchers = [
+      'text',
+      ['text', 'password'],
+      /text/,
+      function (type) {
+        return type === 'text'
+      }
+    ]
+    matchers.forEach(function (matcher) {
+      assert.isTrue(fromDefinition(fixture.firstChild, {
+        properties: {
+          type: matcher
+        }
+      }))
+    })
+    assert.isFalse(fromDefinition(fixture.firstChild, {
+      properties: {
+        type: 'password'
+      }
+    }))
+  })
+
+  it('returns true when all matching properties return true', function () {
+    fixture.innerHTML = '<input value="bar" aria-disabled="true" />'
+    assert.isTrue(fromDefinition(fixture.firstChild, {
+      nodeName: 'input',
+      properties: {
+        type: 'text',
+        value: 'bar'
+      },
+      attributes: {
+        'aria-disabled': 'true'
+      }
+    }))
+  })
+
+  it('returns false when some matching properties return false', function () {
+    fixture.innerHTML = '<input value="bar" aria-disabled="true" />'
+    assert.isFalse(fromDefinition(fixture.firstChild, {
+      nodeName: 'input',
+      attributes: {
+        'aria-disabled': 'false'
+      }
+    }))
+  })
+
+  describe('with virtual nodes', function () {
+    it('matches using a string', function () {
+      fixture.innerHTML = '<div>foo</div>'
+      var node = { actualNode: fixture.firstChild }
+      assert.isTrue(fromDefinition(node, 'div'))
+      assert.isFalse(fromDefinition(node, 'span'))
+    })
+
+    it('matches nodeName', function () {
+      fixture.innerHTML = '<div>foo</div>'
+      var node = { actualNode: fixture.firstChild }
+      assert.isTrue(fromDefinition(node, {
+        nodeName: 'div'
+      }))
+      assert.isFalse(fromDefinition(node, {
+        nodeName: 'span'
+      }))
+    })
+
+    it('matches attributes', function () {
+      fixture.innerHTML = '<div foo="bar">foo</div>'
+      var node = { actualNode: fixture.firstChild }
+      assert.isTrue(fromDefinition(node, {
+        attributes: {
+          foo: 'bar'
+        }
+      }))
+      assert.isFalse(fromDefinition(node, {
+        attributes: {
+          foo: 'baz'
+        }
+      }))
+    })
+
+    it('matches properties', function () {
+      fixture.innerHTML = '<input value="foo" />'
+      var node = { actualNode: fixture.firstChild }
+      assert.isTrue(fromDefinition(node, {
+        properties: {
+          value: 'foo'
+        }
+      }))
+      assert.isFalse(fromDefinition(node, {
+        properties: {
+          value: 'bar'
+        }
+      }))
+    })
+  })
+
+  describe('with a `condition` property', function () {
+    it('calls condition and uses its return value as a matcher', function () {
```

Comment:
may be worth adding a test case where, an attribute is removed from the fixture within a setTimeout, and then an assertion is performed to validate the return value.

Reply:
Never mind this comment, I believe this will work.